src/kagglehub/models_helpers.py

```diff
@@ -0,0 +1,227 @@
+
+    def __enter__(self):
+        self._temp_dir = tempfile.mkdtemp()
+        _, dir_name = os.path.split(self._fullpath)
+        self.path = shutil.make_archive(os.path.join(self._temp_dir, dir_name), self._format, self._fullpath)
+        _, self.name = os.path.split(self.path)
+        return self
+
+    def __exit__(self, *args):
+        shutil.rmtree(self._temp_dir)
+
+
+class File(object):  # noqa: UP004
+    def __init__(self, init_dict):
+        parsed_dict = {k: parse(v) for k, v in init_dict.items()}
+        self.__dict__.update(parsed_dict)
+        self.size = File.get_size(self.totalBytes)
+
+    def __repr__(self):
+        return self.ref
+
+    @staticmethod
+    def get_size(size, precision=0):
+        suffixes = ["B", "KB", "MB", "GB", "TB"]
+        suffix_index = 0
+        while size >= 1024 and suffix_index < 4:  # noqa: PLR2004
+            suffix_index += 1
+            size /= 1024.0
+        return "%.*f%s" % (precision, size, suffixes[suffix_index])
+
+
+def _upload_blob(file_path):
+    data = {
+        "type": "model",
+        "name": os.path.basename(file_path),
+        "contentLength": os.path.getsize(file_path),
+        "lastModifiedEpochSeconds": int(os.path.getmtime(file_path)),
+    }
+    api_client = KaggleApiV1Client()
+    response = api_client.post("/blobs/upload", data=data)
+    if "error" in response and response["error"] != "":
+        raise BackendError(response["error"])
+
+    with open(file_path, "rb") as f:
+        file_data = f.read()
+
+    headers = {"Content-Type": "application/octet-stream"}
+    try:
+        requests.put(response["createUrl"], data=file_data, headers=headers, timeout=10)
+    except requests.exceptions.Timeout:
+        # Handle the timeout exception
+        logger.error("Request timed out")
+
+    if "error" in response and response["error"] != "":
+        raise BackendError(response["error"])
+
+    return response["token"]
+
+
+def upload_files(folder, quiet=False, dir_mode="skip"):  # noqa: FBT002
+    """upload files in a folder
+        Parameters
+    ==========
+    request: the prepared request
+    folder: the folder to upload from
+    quiet: suppress verbose output (default is False)
+    """
+    tokens = []
+    for file_name in os.listdir(folder):
+        if file_name in [
+            DATASET_METADATA_FILE,
+            OLD_DATASET_METADATA_FILE,
+            KERNEL_METADATA_FILE,
+            MODEL_METADATA_FILE,
+            MODEL_INSTANCE_METADATA_FILE,
+        ]:
+            continue
+        tokens.append(_upload_file_or_folder(folder, file_name, dir_mode, quiet))
+    return tokens
+
+
+def _upload_file_or_folder(parent_path, file_or_folder_name, dir_mode, quiet=False):  # noqa: FBT002
+    full_path = os.path.join(parent_path, file_or_folder_name)
+    if os.path.isfile(full_path):
+        return _upload_file(file_or_folder_name, full_path, quiet)
+
+    elif os.path.isdir(full_path):
+        if dir_mode in ["zip", "tar"]:
```

Comment:
for now, only support direct upload (no archive creation) and upload each file inside a dir individually instead.